encoding/codecs/x264.md

```diff
@@ -0,0 +1,292 @@
+It is widely supported for playback in all modern browsers
+and many hardware devices such as gaming consoles and phones.
+It provides better video quality at smaller file sizes
+compared to its predecessors.
+
+x264 is a mature, free, open-source encoder
+for the H.264 video format.
+This guide will focus on x264.
+
+## Prerequisites
+
+To get started, you'll need two things:
+
+- A video to encode--for the examples,
+  we will pipe in a video from VapourSynth,
+  which you should be able to do
+  if you've been following the previous sections of this guide
+- The x264 encoder
+
+Here's how we get a copy of the x264 encoder:
+
+### Windows
+
+Official Windows builds are available
+[here](https://download.videolan.org/pub/x264/binaries/win64/).
+
+### Linux/macOS
+
+Generally, x264 will be available
+through your distribution's package manager.
+Here are a few examples:
+
+- **Ubuntu/Debian**: `sudo apt install x264`
+- **Arch Linux**: `sudo pacman -S x264`
+- **macOS**: `brew install x264`
+
+## Getting Started
+
+x264 is very configurable,
+and the options may seem overwhelming.
+But you can get started encoding
+by using the presets x264 provides
+and understanding a few basic concepts.
+We'll walk through those concepts
+with the following examples.
+
+## Example 1: General-Purpose Encoding
+
+Open up a terminal window,
+and navigate to the folder
+where your VapourSynth script lives.
+Let's run the following command:
+
+```
+vspipe --y4m myvideo.vpy - | x264 --demuxer y4m --preset veryfast --tune animation --crf 24 -o x264output.mkv -
+```
+
+Let's run through what each of these options means:
+
+#### `vspipe --y4m myvideo.vpy -`
+
+This portion loads your VapourSynth script
+and pipes it to stdout,
+adding y4m headers that x264 can decode.
+If you use Linux,
+you're probably familiar with how piping works.
+If you're not,
+it's basically a way of chaining two commands together.
+In this case, we want to chain `vspipe`,
+the program that reads VapourSynth scripts,
+with `x264`, our encoder.
+
+#### `--demuxer y4m`
+
+This tells x264 that we're providing it with a y4m file.
+This matches up with the `--y4m` flag
+that we gave to the `vspipe` command.
+
+#### `--preset veryfast`
+
+x264 has a set of presets
+to switch between faster encoding, or higher quality.
+The full list of presets, from fastest to slowest, is:
+
+1. ultrafast
+1. superfast
+1. veryfast
+1. faster
+1. fast
+1. medium
+1. slow
+1. slower
+1. veryslow
+1. placebo
+
+You will almost never want to use the extreme settings,
+but generally, if you want good quality
+and don't care about how long the encode takes,
+`slower` or `veryslow` are recommended.
+In this example,
+because we are just demonstrating how x264 works,
+we want a fast encode and have chosen `veryfast`.
+
+For the curious,
+you can see a full list of the settings enabled by each preset
+by running `x264 --fullhelp | less` (Linux/Mac)
+or `x264 --fullhelp | more` (Windows).
```

Comment:
We should probably mention that fullhelp isn't fully up-to-date (and it likely won't be updated) in a footnote here.